server/user-groups.ts

```diff
@@ -92,13 +99,14 @@ export abstract class Auth extends Map<ID, GroupSymbol | ''> {
 		user: User,
 		permission: string,
 		target: User | GroupSymbol | null,
-		room?: Room | BasicRoom | null
+		room?: Room | BasicRoom | null,
+		useVisualGroup?: boolean
```

Comment:
I'm not sure the `useVisualGroup` parameters everywhere are necessary. One thing we could do to get around this is to just always unhiderank+retry on permission failure. And then all permissions would be based on visual rank.

Reply:
The issue I see with that is that (from my understanding; I'm probably missing something) non-essential checks would be based on visual group. For example, if an admin hid their rank as a Voice, then tried to use `/alts` on another user, they wouldn't be able to see that user's alts or IP address without unhiding their rank. Is that what you want?